### DISCUSSION TITLE
开发者例会议题收集（2023-02-21）

### DISCUSSION THREAD

**Discussion by chuntaojun** [polarismesh/polaris#915] [Meeting]
# 开发者例会议题收集（2023-02-21）

## 会议信息

andrewshan(单家骏) 邀请您参加腾讯会议
会议主题：PolarisMesh社区双周例会
会议时间：2023/02/21 20:00-20:30 (GMT+08:00) 中国标准时间 - 北京
重复周期：2022/12/12-2023/03/21 20:00-20:30, 每两周 (周二)

点击链接入会，或添加至会议列表：
https://meeting.tencent.com/dm/bi26r1VIAx2y

#腾讯会议：825-9154-7185

###

- 社区 commiter alexwanglei 讨论北极星配置加密的方案
- 北极星 1.14 版回顾 & 北极星 1.15 版本功能特性预告

#### 服务治理

- https://github.com/polarismesh/polaris/issues/935
- https://github.com/polarismesh/polaris/issues/936
- https://github.com/polarismesh/polaris/issues/937
- https://github.com/polarismesh/polaris/issues/938

#### 使用问题

- https://github.com/polarismesh/polaris/issues/892

#### 使用体验

- https://github.com/polarismesh/polaris/issues/922

## 会议录屏

会议录制：PolarisMesh社区双周例会
日期：2023-02-21 19:03:39
录制文件：https://meeting.tencent.com/wework/cloud-record/share?id=bd62c619-877f-402d-8e6f-22e392ed1f05&hide_more_btn=true&from=qywx

访问密码：TWzW

**Comment by alexwanglei** [comment]
优化配置中心逻辑，使其支持类似插件化的扩展能力， 使用middleware/责任链的模式，方便用户灵活添加扩展，首先支持配置加密能力。

**Reply by Guo-Zhang** [reply]
责任链模式的中间件，主要的难点在于用户需要非常小心地维护中间件的顺序。如果中间件比较多并且有可能相互冲突，则需要仔细考虑如何帮助用户排查（异常检测）或者避免（文档和教程）。

配置中心的中间件，主要是在从配置中心到微服务的链路的哪里发挥作用？

**Reply by Guo-Zhang** [reply]
配置加密问题本身，最难处理的是选择加密算法以及设计加密机制。如果使用对称加密，如何配置密钥（手动/自动）确保安全并且方便是个不小的问题。如果使用非对称加密，那么大概机制就是微服务把公钥注册到配置中心，配置中心使用公钥加密以后发给微服务解密。这个注册过程是否需要和微服务注册流程整合，或者以其他方式方便微服务配置？

**Comment by chuntaojun** [comment]
## 该周会相关议题当前进展

#### 服务治理

下列 issue 所提的诉求社区接收，其中涉及官网文档说明补充的将在 2023-03-16 ～ 2023-03-17 进行发布

- https://github.com/polarismesh/polaris/issues/936
- https://github.com/polarismesh/polaris/issues/937


问题 https://github.com/polarismesh/polaris/issues/935、https://github.com/polarismesh/polaris/issues/938 待用户补充具体的使用场景 case 到 issue 中

#### 使用问题

该 issue 对应的问题已将解决方案以及需要解决的场景补充在 issue 的评论中了，社区将会在 1.16 版本中解决这个问题，届时需要使用新的 SDK 以及 北极星服务端

- https://github.com/polarismesh/polaris/issues/892

![Screenshot 2023-03-16 at 14 47 27](https://user-images.githubusercontent.com/23445172/225536866-26b48042-28b3-486a-8aed-41593ab57ca6.png)

#### 使用体验

该功能已有PR提交，预计在 1.16 版本提供，届时用户可以通过调整配置文件中相关任务配置，对应 PR：https://github.com/polarismesh/polaris/pull/976

- https://github.com/polarismesh/polaris/issues/922
